src/main/scala/com/microsoft/hyperspace/index/rankers/JoinIndexRanker.scala

```diff
@@ -35,14 +35,22 @@ object JoinIndexRanker {
    *
    * @param indexPairs index pairs for left and right side of the join. All index pairs are
    *                   compatible with each other.
+   * @param hybridScanEnabled hybridScan config
    * @return rearranged index pairs according to their ranking. The first is the best.
    */
   def rank(
-      indexPairs: Seq[(IndexLogEntry, IndexLogEntry)]): Seq[(IndexLogEntry, IndexLogEntry)] = {
+      indexPairs: Seq[(IndexLogEntry, IndexLogEntry)],
+      hybridScanEnabled: Boolean = false): Seq[(IndexLogEntry, IndexLogEntry)] = {
 
     indexPairs.sortWith {
       case ((left1, left2), (right1, right2)) =>
-        if (left1.numBuckets == left2.numBuckets && right1.numBuckets == right2.numBuckets) {
+        if (hybridScanEnabled && ((left1.allSourceFiles.size + left2.allSourceFiles.size) >
```

Comment:
nit: In my opinion, it is better to document this (and also change the method documentation above - is that still the most up-to-date?). I'm asking since the docs don't really talk about the behavior of the ranker when `hybridScanEnabled = true`.